librz/core/cmd/cmd_flag.c

```diff
@@ -299,115 +280,197 @@ static void spaces_list(RzSpaces *sp, RzOutputMode mode) {
 	}
 }
 
+static int cmpflag(const void *_a, const void *_b) {
+	const RzFlagItem *flag1 = _a, *flag2 = _b;
+	return (flag1->offset - flag2->offset);
+}
+
 RZ_IPI void rz_core_flag_describe(RzCore *core, ut64 addr, bool strict_offset, RzOutputMode mode) {
 	RzFlagItem *f = rz_flag_get_at(core->flags, addr, !strict_offset);
-	if (f) {
-		if (f->offset != addr) {
-			if (mode == RZ_OUTPUT_MODE_JSON) {
-				PJ *pj = pj_new();
-				pj_o(pj);
-				pj_kn(pj, "offset", f->offset);
-				pj_ks(pj, "name", f->name);
-				// Print flag's real name if defined
-				if (f->realname) {
-					pj_ks(pj, "realname", f->realname);
-				}
-				pj_end(pj);
-				rz_cons_println(pj_string(pj));
-				if (pj) {
-					pj_free(pj);
-				}
+	if (!f) {
+		return;
+	}
+	if (f->offset != addr) {
+		if (mode == RZ_OUTPUT_MODE_JSON) {
+			PJ *pj = pj_new();
+			pj_o(pj);
+			pj_kn(pj, "offset", f->offset);
+			pj_ks(pj, "name", f->name);
+			// Print flag's real name if defined
+			if (f->realname) {
+				pj_ks(pj, "realname", f->realname);
+			}
+			pj_end(pj);
+			rz_cons_print(pj_string(pj));
+			if (pj) {
+				pj_free(pj);
+			}
+		} else {
+			// Print realname if exists and asm.flags.real is enabled
+			if (core->flags->realnames && f->realname) {
+				rz_cons_printf("%s + %d\n", f->realname,
+					(int)(addr - f->offset));
 			} else {
-				// Print realname if exists and asm.flags.real is enabled
-				if (core->flags->realnames && f->realname) {
-					rz_cons_printf("%s + %d\n", f->realname,
-						(int)(addr - f->offset));
-				} else {
-					rz_cons_printf("%s + %d\n", f->name,
-						(int)(addr - f->offset));
-				}
+				rz_cons_printf("%s + %d\n", f->name,
+					(int)(addr - f->offset));
+			}
+		}
+	} else {
+		if (mode == RZ_OUTPUT_MODE_JSON) {
+			PJ *pj = pj_new();
+			pj_o(pj);
+			pj_ks(pj, "name", f->name);
+			// Print flag's real name if defined
+			if (f->realname) {
+				pj_ks(pj, "realname", f->realname);
 			}
+			pj_end(pj);
+			rz_cons_print(pj_string(pj));
+			pj_free(pj);
 		} else {
-			if (mode == RZ_OUTPUT_MODE_JSON) {
-				PJ *pj = pj_new();
-				pj_o(pj);
-				pj_ks(pj, "name", f->name);
-				// Print flag's real name if defined
-				if (f->realname) {
-					pj_ks(pj, "realname", f->realname);
-				}
-				pj_end(pj);
-				rz_cons_println(pj_string(pj));
-				pj_free(pj);
+			// Print realname if exists and asm.flags.real is enabled
+			if (core->flags->realnames && f->realname) {
+				rz_cons_println(f->realname);
 			} else {
-				// Print realname if exists and asm.flags.real is enabled
-				if (core->flags->realnames && f->realname) {
-					rz_cons_println(f->realname);
-				} else {
-					rz_cons_println(f->name);
-				}
+				rz_cons_println(f->name);
```

Comment:
Please simplify this.

Reply:
you can set some inline functions for realname and same for the json output

Reply:
Done.